### PR TITLE
switched to `apex-enterprise-patterns/setup-sfdx@v2`

### PR DESCRIPTION
**Pull request by ImJohnMDaniel**
This change is [Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95)

### CODE REVIEW

**Review by daveespo — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @ImJohnMDaniel and @stohn777)

___
*[`.github/workflows/deploy.and.test.yml` line 26 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95#-Nm0qIaq7NakAQYDAgA-:-Nm0qIaq7NakAQYDAgA0:b-pbukp6) ([raw file](https://github.com/apex-enterprise-patterns/force-di/blob/5a8e0943dc91fa1f283be4e225b9e30f5e161e15/.github/workflows/deploy.and.test.yml#L26)):*
> ```YAML
>       # Setup the config parameters needed
>       - run: sf config set target-dev-hub SFDX-ENV --global #Even though the setup-sfdx action uses --setdefaultdevhubusername, it doesn't seem to stick since it uses --setdefaultusername so we brute force it here
>       # Create the scratch org
> ```

What's the reasoning with these rather uninformative comments in the YAML? Comments should only be used for unidiomatic code .. there's no mystery here as to what's going on

If you want to add the `name` property so that the log output in the GH UI is more palatable, that makes more sense ..

**Comment by ImJohnMDaniel** [comment]
> > _[Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95)_ status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @ImJohnMDaniel and @stohn777)
> 
> _[`.github/workflows/deploy.and.test.yml` line 26 at r1](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95#-Nm0qIaq7NakAQYDAgA-:-Nm0qIaq7NakAQYDAgA0:b-pbukp6) ([raw file](https://github.com/apex-enterprise-patterns/force-di/blob/5a8e0943dc91fa1f283be4e225b9e30f5e161e15/.github/workflows/deploy.and.test.yml#L26)):_
> 
> > ```yaml
> >       # Setup the config parameters needed
> >       - run: sf config set target-dev-hub SFDX-ENV --global #Even though the setup-sfdx action uses --setdefaultdevhubusername, it doesn't seem to stick since it uses --setdefaultusername so we brute force it here
> >       # Create the scratch org
> > ```
> 
> What's the reasoning with these rather uninformative comments in the YAML? Comments should only be used for unidiomatic code .. there's no mystery here as to what's going on
> 
> If you want to add the `name` property so that the log output in the GH UI is more palatable, that makes more sense ..

@daveespo, in my opinion, the comments help with the readability; nothing more.

File: .github/workflows/deploy.and.test.yml [hunk 1]
```diff
@@ -15,14 +15,28 @@ jobs:
...
+      - run: sf config set target-dev-hub SFDX-ENV --global #Even though the setup-sfdx action uses --setdefaultdevhubusername, it doesn't seem to stick since it uses --setdefaultusername so we brute force it here
+      # Create the scratch org
+      - run: sf org create scratch --definition-file config/project-scratch-def.json --set-default --duration-days 1 --no-track-source
+      # Deploy/compile the codebase
+      - run: sf project deploy start
+      # Run the core framework tests
+      - run: sf apex run test --wait 5
+      # Intentionally install the Sample Code after the core AEP Commons test pass succeeds so that we don't deploy anything in Sample Code
+      # that could mask a test failure. A much more involved solution would've been to do a workflow_dispatch to the samplecode project.
+      - run: sf shane github src install --convert --githubuser apex-enterprise-patterns --repo force-di-samples --path force-app-1
```
Comment (stohn777) [inline .github/workflows/deploy.and.test.yml:34]: Why are you installing each directory separately? Why not one instance of ...
`sf shane github src install --convert --githubuser apex-enterprise-patterns --repo force-di-samples --path sfdx-source`
Does Shane's plugin not work that way? `sf project deploy start -d sfdx-source/` would do so.

**Review by stohn777 — DISMISSED** [review]

**Review by daveespo — APPROVED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/apex-enterprise-patterns/force-di/95)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ImJohnMDaniel)

**Pull request merged**